src/server/internalServer.cpp

```diff
@@ -571,25 +575,22 @@ std::unique_ptr<Response> InternalServer::handle_search(const RequestContext& re
   try {
     auto searchInfo = SearchInfo(request);
 
-    std::string bookId;
-    if (!searchInfo.bookName.empty()) {
-      try {
-        bookId = mp_nameMapper->getIdForName(searchInfo.bookName);
-      } catch (const std::out_of_range&) {
-        throw std::invalid_argument("The requested book doesn't exist.");
-      }
-    }
-
-
     /* Make the search */
     // Try to get a search from the searchInfo, else build it
     std::shared_ptr<zim::Search> search;
     try {
       search = searchCache.getOrPut(searchInfo,
         [=](){
           std::set<std::string> bookIds;
-          if(!bookId.empty()) {
-            bookIds.insert(bookId);
+          if(!searchInfo.bookNames.empty()) {
+            for(const auto& bookName: searchInfo.bookNames) {
+              try {
+                auto bookId = mp_nameMapper->getIdForName(bookName);
+                bookIds.insert(bookId);
+              } catch(const std::out_of_range&) {
+                throw std::invalid_argument("The requested book doesn't exist.");
+              }
+            }
           } else {
             for (auto& bookId: mp_library->filter(kiwix::Filter().local(true).valid(true))) {
               bookIds.insert(bookId);
```

Comment:
Nesting level has grown too high here. This is a sign for introducing a helper function. IMHO, a private method `std::shared_ptr<zim::Search> Library::createSearch(const SearchInfo&)` is justified.

Reply:
I see that a later commit eliminates this issue.